src/java.base/share/classes/sun/security/provider/SunEntries.java

```diff
@@ -108,6 +110,54 @@ public final class SunEntries {
 
         // common attribute map
         HashMap<String, String> attrs = new HashMap<>(3);
+        if (FIPSConfigurator.enableFips()) {
+            // FIPS supported
+            attrs.put("ImplementedIn", "Software");
+        }
+        /*
+         * Certificates
+         */
+        addWithAlias(p, "CertificateFactory", "X.509",
+                "sun.security.provider.X509Factory", attrs);
+
+        /*
+         * CertStores
+         */
+        add(p, "CertStore", "Collection",
+                "sun.security.provider.certpath.CollectionCertStore",
+                attrs);
+        add(p, "CertStore", "com.sun.security.IndexedCollection",
+                "sun.security.provider.certpath.IndexedCollectionCertStore",
+                attrs);
+
+        /*
+         * Policy
+         */
+        add(p, "Policy", "JavaPolicy", "sun.security.provider.PolicySpiFile");
+
+        /*
+         * Configuration
+         */
+        add(p, "Configuration", "JavaLoginConfig",
+                "sun.security.provider.ConfigFile$Spi");
+
+        /*
+         * CertPathBuilder and CertPathValidator
+         */
+        attrs.clear();
+        attrs.put("ValidationAlgorithm", "RFC5280");
+        attrs.put("ImplementedIn", "Software");
+
+        add(p, "CertPathBuilder", "PKIX",
+                "sun.security.provider.certpath.SunCertPathBuilder",
+                attrs);
+        add(p, "CertPathValidator", "PKIX",
+                "sun.security.provider.certpath.PKIXCertPathValidator",
+                attrs);
+
+        if (FIPSConfigurator.enableFips()) {
+            return;
+        }
 
         /*
```

Comment:
The `attr` map must be cleared here so the following have the same properties when not in FIPS mode.